File: ProjetFinal/src/main/java/factory/model/Formateur.java
package factory.model;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;


import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name = "trainer")
public class Formateur extends RessourcesHumaines {
	@Id
	@GeneratedValue
	@JsonView(Views.ViewCommon.class)
	private Long id;
	@ManyToMany
	@JoinTable(name = "trainer_subject", joinColumns = {@JoinColumn(name = "trainer_id")}, inverseJoinColumns = { @JoinColumn(name = "subject_id")})
	@JsonView(Views.ViewFormateur.class)
	private List<Matiere> matieres = new ArrayList<>();
	@OneToOne
	@JsonView(Views.ViewFormateurWithFormations.class)
	@JoinColumn(name = "training_id")
	private Formation formation;
	@OneToOne(mappedBy = "formateur")
	@JsonView(Views.ViewFormateurWithModules.class)
	private Module module;

	
	/**** CONSTRUCTEURS*/
	
	public Formateur() {
		super();
	}

	public Formateur(String nom, String prenom, String coordonnees, String rue, String ville, String codePostal,
			Boolean accesAdministrateur, Boolean accesFormateur, Boolean accesStagiaire, Boolean accesTechnicien,
			String username, String password) {
		super(nom, prenom, coordonnees, rue, ville, codePostal, accesAdministrateur, accesFormateur, accesStagiaire,
				accesTechnicien, username, password);
	}

	

	public Formateur(String nom, String prenom, String coordonnees, String rue, String ville, String codePostal,
			Boolean accesAdministrateur, Boolean accesFormateur, Boolean accesStagiaire, Boolean accesTechnicien,
			String username, String password, List<Matiere> matieres, Formation formation, Module module) {
		this(nom, prenom, coordonnees, rue, ville, codePostal, accesAdministrateur, accesFormateur, accesStagiaire,
				accesTechnicien, username, password);
		this.matieres = matieres;
		this.formation = formation;
		this.module = module;
	}

	/**** METHODES*/	

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public List<Matiere> getMatieres() {
		return matieres;
	}

	public void setMatieres(List<Matiere> matieres) {
		this.matieres = matieres;
	}

	public Formation getFormation() {
		return formation;
	}

	public void setFormation(Formation formation) {
		this.formation = formation;
	}

	public Module getModule() {
		return module;
	}

	public void setModule(Module module) {
		this.module = module;
	}

	

}

File: ProjetFinal-angular/src/app/model/formateur.ts
export class Formateur {
  constructor(private _id: number, private _nom: string, private _prenom: string,
              private _coordonnees: string, private _rue: string, private _ville: string,
              private _codePostal: string, private _username: string, private _password: string,
              private _accesAdministrateur: boolean, private _accesFormateur: boolean,
              private _accesStagiaire: boolean, private _accesTechnicien: boolean) {
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get nom(): string {
    return this._nom;
  }

  set nom(value: string) {
    this._nom = value;
  }

  get prenom(): string {
    return this._prenom;
  }

  set prenom(value: string) {
    this._prenom = value;
  }

  get coordonnees(): string {
    return this._coordonnees;
  }

  set coordonnees(value: string) {
    this._coordonnees = value;
  }

  get rue(): string {
    return this._rue;
  }

  set rue(value: string) {
    this._rue = value;
  }

  get ville(): string {
    return this._ville;
  }

  set ville(value: string) {
    this._ville = value;
  }

  get codePostal(): string {
    return this._codePostal;
  }

  set codePostal(value: string) {
    this._codePostal = value;
  }

  get username(): string {
    return this._username;
  }

  set username(value: string) {
    this._username = value;
  }

  get password(): string {
    return this._password;
  }

  set password(value: string) {
    this._password = value;
  }

  get accesAdministrateur(): boolean {
    return this._accesAdministrateur;
  }

  set accesAdministrateur(value: boolean) {
    this._accesAdministrateur = value;
  }

  get accesFormateur(): boolean {
    return this._accesFormateur;
  }

  set accesFormateur(value: boolean) {
    this._accesFormateur = value;
  }

  get accesStagiaire(): boolean {
    return this._accesStagiaire;
  }

  set accesStagiaire(value: boolean) {
    this._accesStagiaire = value;
  }

  get accesTechnicien(): boolean {
    return this._accesTechnicien;
  }

  set accesTechnicien(value: boolean) {
    this._accesTechnicien = value;
  }
}
